components/datetime/src/time_zone.rs

```diff
@@ -269,12 +270,113 @@ impl<'data> TimeZoneFormat<'data> {
         Ok(time_zone_format)
     }
 
+    pub fn try_from_config<L, ZP>(
+        locale: L,
+        config: TimeZoneFormatConfig,
+        zone_provider: &ZP,
+    ) -> Result<Self, DateTimeFormatError>
+    where
+        L: Into<Locale>,
+        ZP: DataProvider<'data, provider::time_zones::TimeZoneFormatsV1Marker>
+            + DataProvider<'data, provider::time_zones::ExemplarCitiesV1Marker>
+            + DataProvider<'data, provider::time_zones::MetaZoneGenericNamesLongV1Marker>
+            + DataProvider<'data, provider::time_zones::MetaZoneGenericNamesShortV1Marker>
+            + DataProvider<'data, provider::time_zones::MetaZoneSpecificNamesLongV1Marker>
+            + DataProvider<'data, provider::time_zones::MetaZoneSpecificNamesShortV1Marker>
+            + ?Sized,
+    {
+        let locale = locale.into();
+
+        let zone_formats: DataPayload<TimeZoneFormatsV1Marker> = zone_provider
+            .load_payload(&DataRequest {
+                resource_path: ResourcePath {
+                    key: provider::key::TIMEZONE_FORMATS_V1,
+                    options: ResourceOptions {
+                        variant: None,
+                        langid: Some(locale.clone().into()),
+                    },
+                },
+            })?
+            .take_payload()?;
+
+        let mut time_zone_format = Self {
+            config: Some(config),
+            patterns: None,
+            zone_formats,
+            exemplar_cities: None,
+            mz_generic_long: None,
+            mz_generic_short: None,
+            mz_specific_long: None,
+            mz_specific_short: None,
+        };
+
+        match time_zone_format
+            .config
+            .expect("TimeZoneFormat::try_from_config() expected Some(config) but got None")
```

Comment:
Suggestion: the config enum can be `Copy` and you can just copy it so you don't have to do the `expect()`.